bumpversion/cli.py

```diff
@@ -650,7 +650,10 @@ def _update_config_file(
 
         if write_to_config_file:
             with open(config_file, "wt", encoding="utf-8", newline=config_newlines) as f:
-                f.write(new_config.getvalue().strip() + "\n")
+                config_lines = new_config.getvalue().strip().split("\n")
+                # prevent tailing white spaces
+                config_lines = [line.rstrip(" ") + "\n" for line in config_lines]
```

Comment:
You only test for spaces here. Why? Aren't tabs also important?

Reply:
While traiking tabs are possible in a hand written `setup.cfg`, this is about `configparser` writing the parsed and modified config back to a string.
I did only see trainling spaces being added by configparser and that only happen for lines which end with `=`.
Spaces used for indentation, on the other hand, are being converted to tabs.

Reply:
I also added tests for trailing tabs and my usecase to the test.

Reply:
Might want to split on `os.linesep` rather than `\n` here?

Reply:
@bsolomon1124 `os.linesep` wouldn't always work e.g. if the file has a `\n` line ending (set by .editorconfig or .gitattributes) and the OS is windows with `\r\n` as `linesep`.
Also, I found that python does line ending sanitizing when reading in text mode (see https://github.com/c4urself/bump2version/pull/161#issuecomment-773897993).